From d437ecf75de2d6fdeb2aed6f45c4b3b16373389b Mon Sep 17 00:00:00 2001
From: Sergey Poznyakoff <gray@gnu.org>
Date: Fri, 11 Aug 2023 21:35:30 +0300
Subject: [PATCH] Revert "Fix savannah bug #63567"
Bug-Fedora: https://bugzilla.redhat.com/show_bug.cgi?id=2230127
Bug-Debian: https://bugs.debian.org/748244

Commit e89c7a45eb broke deletion from archives. The reported number
of bytes read is rounded to the nearest record anyway, revert the
commit and document the fact.

Reported by Ed Santiago. See
https://bugzilla.redhat.com/show_bug.cgi?id=2230127

* doc/tar.texi: Document the fact that --totals rounds up the
number of bytes reads to the nearest record.
* src/buffer.c: Revert changes.
* tests/delete06.at: Fix expected status code and stderr.
---
 doc/tar.texi      | 5 +++++
 src/buffer.c      | 3 +--
 tests/delete06.at | 7 +++++--
 3 files changed, 11 insertions(+), 4 deletions(-)

Index: tar/src/buffer.c
===================================================================
--- tar.orig/src/buffer.c
+++ tar/src/buffer.c
@@ -987,8 +987,7 @@ short_read (size_t status)
     }
 
   record_end = record_start + (record_size - left) / BLOCKSIZE;
-  if (left == 0)
-    records_read++;
+  records_read++;
 }
 
 /*  Flush the current buffer to/from the archive.  */
Index: tar/tests/delete06.at
===================================================================
--- tar.orig/tests/delete06.at
+++ tar/tests/delete06.at
@@ -36,7 +36,10 @@ esac
 dd if=archive.tar of=trunc.tar bs=$size count=1 2>/dev/null
 tar --delete 'b/' -f trunc.tar
 ],
-[0],
-[],[],[],[],[gnu, pax])
+[2],
+[],
+[tar: lseek: trunc.tar: Value too large for defined data type
+tar: Exiting with failure status due to previous errors
+],[],[],[gnu, pax])
 
 AT_CLEANUP
